diesel/src/lib.rs

```diff
@@ -547,3 +547,99 @@ pub use crate::result::Error::NotFound;
+//         }
+//     }
+
+//     type TestConnection = PgConnection;
+//     //type TestConnection = MysqlConnection;
+
+//     fn get_test_connection() -> TestConnection {
+//           let url = "postgres://localhost/diesel_tests";
+//         //let url = "mysql://weiznich:diesel@localhost/diesel_tests";
+
+//         let mut conn = TestConnection::establish(url).unwrap();
+//         conn.begin_test_transaction();
+//         conn
+//     }
+
+//     #[derive(AsChangeset)]
+//     #[diesel(table_name = users)]
+//     struct UserChange {
+//         id: i32,
+//         name: Option<String>,
+//     }
+
+//     #[test]
+//     fn test_do_nothing() {
+//         let mut conn = get_test_connection();
+
+//         crate::insert_into(users::table)
+//             .values((users::id.eq(42), users::name.eq("John")))
+//             .execute(&mut conn)
+//             .unwrap();
+
+//         let query = crate::insert_into(users::table)
+//             .values((users::id.eq(42), users::name.eq("John")))
+//             .on_conflict(users::id)
+//             .do_update()
+//             .set(users::name.eq("Other John"));
+//         println!(
+//             "{}",
+//             crate::debug_query::<<TestConnection as Connection>::Backend, _>(&query)
+//         );
+//         query.execute(&mut conn).unwrap();
+
+//         let user = users::table
+//             .find(42)
+//             .select(users::name)
+//             .get_result::<String>(&mut conn)
+//             .unwrap();
+//         assert_eq!(user, "Other John");
+
+//         let query = crate::insert_into(users::table)
+//             .values((users::id.eq(42), users::name.eq("John")))
+//             .on_conflict(users::id)
+//             .do_nothing();
+//         println!(
+//             "{}",
+//             crate::debug_query::<<TestConnection as Connection>::Backend, _>(&query)
+//         );
+//         query.execute(&mut conn).unwrap();
+
+//         let query = crate::insert_into(users::table)
+//             .values((users::id.eq(42), users::name.eq("John")))
+//             .on_conflict_do_nothing();
+//         println!(
+//             "{}",
+//             crate::debug_query::<<TestConnection as Connection>::Backend, _>(&query)
+//         );
+//         query.execute(&mut conn).unwrap();
+
+//         let query = crate::insert_into(users::table)
+//             .values((users::id.eq(42), users::name.eq("John")))
+//             .on_conflict(users::id)
+//             .do_update()
+//             .set(UserChange { id: 42, name: None });
+//         println!(
+//             "{}",
+//             crate::debug_query::<<TestConnection as Connection>::Backend, _>(&query)
+//         );
+//         query.execute(&mut conn).unwrap();
+
+//         let user = users::table
+//             .find(42)
+//             .select(users::name)
+//             .get_result::<String>(&mut conn)
+//             .unwrap();
+//         assert_eq!(user, "Other John");
+//     }
+// }
```

Comment:
Hello, is there a reason why this test is commented on?

Reply:
Good catch. That's a test that I locally used for faster turn around times. It's not supposed to be there at all. I've removed it from the PR.